Tests/test_parameter.py

```diff
@@ -3,93 +3,276 @@
+        with self.assertRaisesRegexp(ValueError, f"parameter_value must be a float or int"):
+            Parameter(parameter_name="None", parameter_value=None)
 
-    def test__get_field_names(self):
+        # test invalid value string
+        with self.assertRaisesRegexp(ValueError, f"No valid parameter value! Accepted format"):
+            Parameter(parameter_name="None", parameter_value='2ba')
+
+        # test string parameter values
+        self.assertTrue(Parameter(parameter_name="None", parameter_value="1.0").value == 1.0)
+
+        self.assertTrue(Parameter(parameter_name="None", parameter_value="1/2").value == 0.5)
+
+        self.assertTrue(Parameter(parameter_name="None", parameter_value="1e2").value == 100)
+
+        # testing invalid parameter name
+        with self.assertRaisesRegexp(ValueError, f"parameter_name should be at least one character and cannot start with a number!"):
+            Parameter(parameter_name="2", parameter_value=2)
+
+    def test_parameter_entry(self):
+        #Valid ParameterEntry Construction
+        ParameterEntry(parameter_name="None", parameter_value=1.0, parameter_key = {"part_id":"id"}, parameter_info = {"comment":"comment"})
+
+        #Invalid keys
+        param_keys = Parameter(parameter_name="None", parameter_value=1.0)
+        with self.assertRaisesRegexp(ValueError, "parameter_key must be"):
+            ParameterEntry(parameter_name="None", parameter_value=1.0, parameter_key = param_keys)
+
+        #Invalid info
+        param_info = "blah blah"
+        with self.assertRaisesRegexp(ValueError, f"parameter_info must be None or a dictionary"):
+            ParameterEntry(parameter_name="None", parameter_value=1.0, parameter_info = param_info)
+
+    def test_model_parameter(self):
+        #valid ModelParameter Construction
+        ModelParameter(parameter_name="None", parameter_value=1.0, search_key = ("that", "this", "k"), found_key = ("this", None, "k"))
+
+        #Invalid keys
+        k = Parameter(parameter_name="None", parameter_value="1.0")
+        with self.assertRaisesRegexp(ValueError,"parameter_key must be None"):
+            ModelParameter(parameter_name="None", parameter_value=1.0, search_key = k, found_key = ("this", None, "k"))
+
+        with self.assertRaisesRegexp(ValueError,"parameter_key must be None"):
+            ModelParameter(parameter_name="None", parameter_value=1.0, search_key = ("that", "this", "k"), found_key = k)
+
+
+    def test_get_field_names(self):
+        PD = ParameterDatabase
         test_accepted_field_names = {'mechanism': ['mechanism', 'mechanism_id']}
         valid_field_names = ['part_id']
 
         # test None as field_names
         with self.assertRaisesRegexp(ValueError, 'field_names must be a list of strings'):
-            Parameter._get_field_names(field_names=None, accepted_field_names=test_accepted_field_names)
+            PD._get_field_names(field_names=None, accepted_field_names=test_accepted_field_names)
```

Comment:
I'm not sure why you call a static method through the object (PD) and not through the class (Parameter) as before. Is there a reason for that?

Reply:
In my mind, ParameterDatabase loads files, Parameter just stores the data.